zerver/views/streams.py

```diff
@@ -602,6 +608,7 @@ def add_subscriptions_backend(
         )
         stream_dict_copy["can_remove_subscribers_group"] = can_remove_subscribers_group
 
+        stream_dict_copy["push_notifications_enabled"] = push_notifications_enabled
         stream_dicts.append(stream_dict_copy)
 
     # Validation of the streams arguments, including enforcement of
```

Comment:
You would also need to make sure that non-admin users cannot set `push_notifications_enabled` as `True` similar to what we do for non-owners to not allow them setting `message_retention_days` other than `None`.